build/build.go

```diff
@@ -61,6 +61,7 @@ type Options struct {
 
 	Ref           string
 	Allow         []entitlements.Entitlement
+	Annotations   map[exptypes.AnnotationKey]string
```

Comment:
The annotations should already exist in the `Exports` array. Can we make a helper function to extract it from there instead of new field.

Reply:
I'm not sure about this. In `Exports` they've all had their keys stringified into e.g. `annotation-index` to fit into the `Attrs` map, so we'd need to re-parse those keys if we wanted to pull them back out. I agree the duplication doesn't feel great but at least this way it's the same parsing function twice from the same source, rather than parsing once, serialising and parsing agian.

Reply:
I think it is correct though. The serialized format is the format the builder accepts and this multi-node merge case is a special case where we are doing part of the build ourself. If/when buildkit supported multiple nodes then we would not do the client side and only sent the `Exports` to buildkit. Maybe some part of the parsing code from BuildKit can be reused in here to reduce code.

Reply:
Ok, I see what you mean. I'll take care of it

Reply:
hopefully sorted